core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -252,21 +252,19 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     Path tablePath = new Path(defaultWarehouseLocation(identifier));
     TableOperations ops = newTableOps(identifier);
-    TableMetadata lastMetadata;
-    if (purge && ops.current() != null) {
-      lastMetadata = ops.current();
-    } else {
-      lastMetadata = null;
-    }
-
+    TableMetadata lastMetadata = ops.current();
     try {
-      if (purge && lastMetadata != null) {
-        // Since the data files and the metadata files may store in different locations,
-        // so it has to call dropTableData to force delete the data file.
-        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+      if (lastMetadata == null) {
+        LOG.error("Not an iceberg table: %s", identifier);
```

Comment:
This is not an error. The API states that if the table doesn't exist, then the catalog should return false. If you want to add a debug log, that's fine. But this should not be info or error level.

Reply:
The reason I thought of putting it in error was calling iceberg api drop on a table which is not an iceberg table looked like an error but wasn't sure of this. But now I agree with you, so have made the log as debug now.